packages/block-library/src/block/edit.js

```diff
@@ -272,17 +306,15 @@ export default function ReusableBlockEdit( {
 
 	return (
 		<RecursionProvider uniqueId={ ref }>
-			<InspectorControls>
-				<PanelBody>
-					<TextControl
-						label={ __( 'Name' ) }
-						value={ title }
-						onChange={ setTitle }
-						__nextHasNoMarginBottom
-						__next40pxDefaultSize
-					/>
-				</PanelBody>
-			</InspectorControls>
+			{ userCanEdit && editOriginalProps && (
+				<BlockControls>
+					<ToolbarGroup>
+						<ToolbarButton { ...editOriginalProps }>
```

Comment:
I wonder if it makes sense to just register this on each editor. For `edit-post` it's a button that will toggle the focus mode, for `edit-site` it's a link that calls `router.navigate()`. This way we avoid introducing the `getPostLinkProps` setting, it's not really related to the `block-library` package anyway. I still have mixed feelings about this though 😅 .

Reply:
The [plan is to get rid of focus mode](https://github.com/WordPress/gutenberg/pull/56827#issuecomment-1856529519). The preference is to move to this approach of passing the new post id and post type to the entity provider.